TSRM/tsrm_win32.c

```diff
@@ -706,7 +690,7 @@ TSRM_API void *shmat(int key, const void *shmaddr, int flags)
 		return (void*)-1;
 	}
 
-	shm->addr = MapViewOfFileEx(shm->segment, FILE_MAP_ALL_ACCESS, 0, 0, 0, NULL);
+	shm->addr = shm->descriptor + sizeof(shm->descriptor);
```

Comment:
The error handling here is wrong, `if (NULL == shm->addr) {` would be comparing `shm->descriptor + sizeof(shm->descriptor);` against NULL

As a result, this would start returning non-null.

At a quick glance, the rest of the pointer operations seemed normal

I suspect some part of this change is related to https://github.com/krakjoe/apcu/issues/445 - apcu calls shmat in 3 places https://github.com/krakjoe/apcu/blob/v5.1.22/apc_shm.c#L72